### DISCUSSION TITLE
Can I operate many accounts with just one API ID & HASH?

### DISCUSSION THREAD

**Discussion by AlienegraGeek** [gotd/td#1346] [Q&A] [answered]
### Description

I want to be able to operate many accounts with just one API ID & HASH. 

For example, I use the API ID of A phone number to get a client, and I want to use this client to manipulate B phone number join channel or send message.

If possible, can you tell me how to do it, because I see that almost all apis are called through the client rather than the phone number.If not, should each user have their own api id？

Thanks.

**Comment by ernado** [comment, marked as answer]
You can reuse app hash and app id between multiple instances, yes.
For example, telegram desktop has a single id-hash pair that is reused between all clients.

**Reply by AlienegraGeek** [reply]
I got it, does that mean that if I want to operate multiple accounts on just one device, even with a single id-hash, I must have to run multiple new client because each account needs an instance？
